include/seng499/msrc_file_system_depth_device.hpp

```diff
@@ -0,0 +1,37 @@
+/** 
+ *	\file
+ */
+
+
+#pragma once
+
+
+#include <seng499/file_system_depth_device.hpp>
+#include <Eigen/Dense>
+#include <regex>
+
+
+namespace seng499 {
+	
+	
+	class msrc_file_system_depth_device_filter: public file_system_depth_device_filter {
+		
+		
+		public:
+			
+			virtual bool operator () (const boost::filesystem::path &) const override;
+				
+	};
+	
+	class msrc_file_system_depth_device_frame_factory: public file_system_depth_device_frame_factory {
+		
+		virtual std::vector<float> operator () (const boost::filesystem::path & path, std::vector<float> vec);
+		
+		virtual std::size_t width () const noexcept;
+		
+		virtual std::size_t height() const noexcept;
+		
+		virtual Eigen::Matrix3f k () const noexcept;
```

Comment:
It just tells the compiler that you mean for the `virtual` method to override one in the base class.

If you typed a parameter wrong or mistyped the name it'd just create a new `virtual` function rather than overriding.  `override` informs the compiler of your intention and forces it to emit a diagnostic if it doesn't override.

It's just considered best practice since it was added in C++11.